src/main/java/org/hyperskill/hstest/stage/SpringTest.java

```diff
@@ -172,12 +181,34 @@ public static void startSpring() throws Exception {
                                 .map(Class::getCanonicalName)
                                 .collect(Collectors.joining(", "))
                 );
+            } else if (!ReflectionUtils.hasMainMethod(suitableClasses.get(0)) && !isKotlin) {
+                throw new ErrorWithFeedback("The main method was not found in the class");
             }
 
-            Class<?> springClass = suitableClasses.get(0);
-            Method mainMethod = ReflectionUtils.getMainMethod(springClass);
-            mainMethod.invoke(null, new Object[] {args});
-            springRunning = true;
+            if (!isKotlin) {
+                ReflectionUtils.getMainMethod(suitableClasses.get(0))
+                        .invoke(null, new Object[]{args});
+                springRunning = true;
+            } else {
+                List<Class<?>> allClassesFromPackage = ReflectionUtils.getAllClassesFromPackage("");
+                allClassesFromPackage.forEach(it -> {
+                    if (it.getCanonicalName().contains("ApplicationKt")) {
```

Comment:
In Kotlin, is it really necessary to name the class/file as `"Application"`? Or it is just the way we currently test Kotlin Spring Boot projects?

Reply:
In the initial project template, the author of the project specifies the name of the class/file, the user in this case does not create it, and therefore should not change it either.

Reply:
Actually, it's easy for the user to change it, nothing stops him, even text `Don't rename the "Application.kt" file`. If Kotlin Spring Boot applications allow for class/file to be named differently, then `hs-test` should support that.

One of the most important ideas in `hs-test` is: "If user runs this application from IDEA and everything works as expected THEN `hs-test` should be able to test this application the same way". So, if the user renames the file with main method and everything still works OK for him (application runs and works like in the description of the stage), then `hs-test` should adapt and test smoothly such user's project.

In case we no way can run user's application (for example, there are two classes with annotation and main method) then we should say exactly why we can't run user's application with names of such two classes and a hint to remove annotation from one of the classes, so the user could change his code immediately to let `hs-test` test their application.

Sorry for a lot of words)

Reply:
Thank you for explaining. I made changes, now hs-test can support any file name with the main method, the key remains the presence of the Kt sufix.